internal/driver/glfw/scale.go

```diff
@@ -0,0 +1,61 @@
+package glfw
+
+import (
+	"math"
+	"os"
+	"strconv"
+
+	"fyne.io/fyne"
+)
+
+const (
+	baselineDPI = 120.0
+	scaleEnvKey = "FYNE_SCALE"
+)
+
+func calculateDetectedScale(widthMm, widthPx int) float32 {
+	dpi := float32(widthPx) / (float32(widthMm) / 25.4)
+	if dpi > 1000 || dpi < 10 {
+		dpi = baselineDPI
+	}
+
+	scale := float32(float64(dpi) / baselineDPI)
+	if scale < 1.0 {
+		return 1.0
+	}
+	return scale
+}
+
+func calculateScale(user, system, detected float32) float32 {
+	if user == fyne.SettingsScaleAuto {
+		user = 1.0
+	}
+
+	if system == fyne.SettingsScaleAuto {
+		system = detected
+	}
+
+	raw := system * user
+	return float32(math.Round(float64(raw*10.0))) / 10.0
```

Comment:
The `.0` is not necessary here because the compiler infers the numbers’ types.

Reply:
May be unnecessary but I like having decimal places when talking about floats :)